src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java

```diff
@@ -332,4 +333,26 @@ private StudentProfilePage getProfilePageForStudent(String studentId) {
         return loginAdminToPage(profileUrl, StudentProfilePage.class);
     }
 
+    @AfterClass
+    public void classTearDown() {
+        // The data associated with the test accounts have to be removed when the test ends because
+        // the test accounts are shared across tests. In particular, the courses of test student1 account is not the same as
+        // `StudentHomePageUiTest` so it has to be cleared before running that test.
+
+        // Test data is never cleared after test in order to save datastore costs as we already remove and restore the data
+        // bundle when the test runs. This raises the question of why we do not remove the data bundle when the class
+        // ends then we do not have to remove it from startup. The reason is that in the case a test fails to tear down
```

Comment:
Sorry, I think we missed out an important point here.

`The reason is that if a test's data is never accessed in another test, there is no need to clear the test's data to prevent incurring additional datastore costs. Also although unlikely in normal conditions, when a test fails to tear down (fail to remove data bundle), another test should have no reason to fail.`

Reply:
Let's also add the relevant information https://github.com/TEAMMATES/teammates/pull/8875#discussion_r200800204 @damithc mentioned.